Refactor mine placer

bcarnevale-myob opened the pull request:
I am stuck on the constructor for PredictableRandom what I should be doing here.

For context Sol, I am pulling out the Random value in the constructor for RandomMinePlacer to remove dependency injection. Have I used that correctly?

bcarnevale-myob committed:
Simplify MinePlacer interface

The MinePlacer interface was confusing because it allowed for incorrect input eg more mines than the field can take. This simplifies the interface to remove the parameter and uses the field size given in the constructor to determine the maximum amount of mines to place.

This requires updating the unit test to remove the parameter.

bcarnevale-myob committed:
Add the IRandom interface to be used in MinePlacer

The IRandom interface will remove dependency injection from the RandomMinePlacer constructor.

bcarnevale-myob committed:
Add RealRandom class which implements IRandom interface

The RealRandom class generates a random number and can call the nextInt method.

bcarnevale-myob committed:
Add PredictableRandom class which implements IRandom

The PredictableRandom class will be used to mock the randomNumberGenerator used in the MinePlacer class.

nskoufis commented on src/Field/PredictableRandom.java:
```diff
@@ -0,0 +1,20 @@
+
+    private int callCounter = 0;
+
+    public PredictableRandom() {
```
I would do something along the lines of
```
public PredictableRandom(int[] intsToReturn)
```

bcarnevale-myob committed:
Add a constructor to PredictableRandom

The constructor will take in an array of ints to be used for testing the RandomMinePlacer.

The PredictableRandom constructor will take in an array of inte

bcarnevale-myob committed:
Test the correct value is passed into nextInt

This test will assert that the correct value has been passed into
nextInt when numberOfMines is called. The value should be the fieldSize
- 1.

bcarnevale-myob committed:
Test the value generated when calling numberOfMines

This test asserts that when the numberOfMines is generated, the result
will be the generated value + 1.

nskoufis reviewed (approved):
Looking really good! Only thing is to remove that `bianca` file that you accidentally commited.

bcarnevale-myob committed:
Removed bianca file
